File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/interfaceInJava/fieldsInInterface/Car.java
package de.fraunhofer.iem.classesAndObjects.interfaceInJava.fieldsInInterface;

/**
 * Demonstrates the implementing the interface that has fields
 *
 * @author Ranjith Krishnamurthy
 */
public class Car implements Vehicle {
    String modelNumber;
    String vehicleRegistrationNumber;
    private boolean isStart = false;

    public Car(String modelNumber, String vehicleRegistrationNumber) {
        this.modelNumber = modelNumber;
        this.vehicleRegistrationNumber = vehicleRegistrationNumber;
    }

    @Override
    public void start() {
        if (isStart) {
            System.err.println("Car is already started.");
        }

        isStart = true;
    }

    @Override
    public void stop() {
        if (!isStart) {
            System.err.println("Car is already stopped.");
        }

        isStart = false;
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/interfaceInKotlin/propertiesInInterface/Car.kt
package de.fraunhofer.iem.classesAndObjects.interfaceInKotlin.propertiesInInterface

/**
 * Demonstrates the implementing the interface that has properties
 *
 * @author Ranjith Krishnamurthy
 */
class Car(override val modelNumber: String, override var vehicleRegistrationNumber: String) : Vehicle {
    private var isStart = false

    override fun start() {
        if (isStart) {
            println("Car is already started.")
        }

        isStart = true
    }

    override fun stop() {
        if (!isStart) {
            println("Car is already stopped.")
        }

        isStart = false
    }
}